src/TestItemRunner.jl

```diff
@@ -124,17 +124,18 @@ function run_tests(path; filter=nothing)
 
     # Filter @testitems
     if filter !== nothing
-        for file in keys(testitems)     
+        for file in keys(testitems)
             testitems[file] = Base.filter(i -> filter((filename=file, name=i.name, tags=i.option_tags)), testitems[file])
+            isempty(testitems[file]) && pop!(testitems, file)
```

Comment:
without this change, the output of TestItemRunner.jl's own tests seemed a little odd to me:
```julia
 Test Summary:                                       | Pass  Total
Package                                             |    9      9
  packages/TestItemDetection/test/test_detection.jl |       No tests
  src/TestItemRunner.jl                             |    9      9
    compute_line_column                             |    9      9
     Testing TestItemRunner tests passed 
```

The `packages/TestItemDetection/test/test_detection.jl` has `@testitem`s (which is why it's getting picked up), but they're not being run because we're filtering them out (we're running with `@run_package_tests filter=i->endswith(i.filename, "TestItemRunner.jl")`) 

So i changed it to not create this testset, and now we get 
```julia
Test Summary:           | Pass  Total  Time
Package                 |    9      9  0.1s
  src/TestItemRunner.jl |    9      9  0.1s
    compute_line_column |    9      9  0.0s
     Testing TestItemRunner tests passed
```

Reply:
Yes, that is better. _Or_ we could mark them as `skipped` or something like that? But we could add that in a new PR later, this here just looks like a straight up improvement as is.